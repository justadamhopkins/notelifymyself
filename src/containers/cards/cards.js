import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Card } from '../../components/card'
import { removeCardEle } from '../../store/actions/card/cardActions'

export class Cards extends Component {
  deleteCard = (event) => {
    const { parentElement } = event.target
    const { id } = parentElement.parentElement.dataset
    const { removeCardEle } = this.props
    removeCardEle(id)
  }

  render() {
    const { cards } = this.props
    return (
    <div>
      <Card
      card={cards}
      deletedCard={this.deleteCard}/>
    </div>
    )
  }
}

Cards.propTypes = {
  cards: PropTypes.object,
  removeCardEle: PropTypes.func
}

export default connect(
  (state) => ({
    cards: state.card
  }),
  { removeCardEle }
)(Cards)
