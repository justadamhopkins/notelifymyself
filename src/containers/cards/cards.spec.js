import React from 'react'
import { mount } from 'enzyme'
import { Cards } from './cards'


describe('<Cards/>', () => {
  it('Should render component', () => {
    const cards = mount(<Cards/>)
    expect(cards).toMatchSnapshot()
  })
})
