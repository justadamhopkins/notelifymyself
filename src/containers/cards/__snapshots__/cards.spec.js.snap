// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards/> Should render component 1`] = `
<Cards
  cards={
    Object {
      "11": Object {
        "id": "11",
        "note": "i love you",
        "pictures": "animals",
        "subtitle": "adam",
        "title": "mr",
      },
    }
  }
  deleteCard={[MockFunction]}
  fetchCards={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  removeCardEle={[MockFunction]}
>
  <div>
    <Card
      card={
        Object {
          "11": Object {
            "id": "11",
            "note": "i love you",
            "pictures": "animals",
            "subtitle": "adam",
            "title": "mr",
          },
        }
      }
      deletedCard={[Function]}
    >
      <div
        className="Card"
      >
        <ul>
          <CardList
            deletedCard={[Function]}
            id="11"
            key="11"
            note="i love you"
            pictures="animals"
            subtitle="adam"
            title="mr"
          >
            <li
              className="Cardlist"
              data-id="11"
            >
              <CloseIcon
                deletedCard={[Function]}
              >
                <button
                  className="close"
                  onClick={[Function]}
                >
                  <img
                    height="30"
                    src=""
                    width="30"
                  />
                </button>
              </CloseIcon>
              <CardTitle
                pictures="animals"
                subtitle="adam"
                title="mr"
              >
                <div
                  className="CardTitle"
                >
                  <div
                    className="CardImage"
                  >
                    <img
                      src="https://placeimg.com/300/200/animals"
                    />
                  </div>
                  <div>
                    <h5>
                      mr
                    </h5>
                    <p>
                      adam
                    </p>
                  </div>
                </div>
              </CardTitle>
              <CardImage
                pictures="animals"
              >
                <div
                  className="CardImage"
                >
                  <img
                    src="https://placeimg.com/300/200/animals"
                  />
                </div>
              </CardImage>
              <CardPara
                note="i love you"
              >
                <div
                  className="CardPara"
                >
                  <p>
                    i love you
                  </p>
                </div>
              </CardPara>
            </li>
          </CardList>
        </ul>
      </div>
    </Card>
  </div>
</Cards>
`;
